bundles/net.tourbook/src/net/tourbook/map/IMapView.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (C) 2024 Frédéric Bard
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110, USA
+ *******************************************************************************/
+package net.tourbook.map;
+
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.swt.graphics.Image;
+
+public interface IMapView {
+
+   Image getMapViewImage();
+
+   void onSelectionChanged(final ISelection selection);
+
+   void restoreState();
+
+   void saveState();
```

Comment:
Those methods in the interface are used in all the maps, hence why I chose to put them in the interface

This is a standard practice when doing [ object-oriented programming](https://en.wikipedia.org/wiki/Interface_(object-oriented_programming))

![image](https://github.com/mytourbook/mytourbook/assets/25235665/e6272891-e017-4b68-bf93-1d2c17382d5c)

Reply:
When I see an interface then I think the methods are used from outside of the java file but this is not for

```
onSelectionChanged(final ISelection selection);
void restoreState();
void saveState();
```

Reply:
In older cases you have reduced the scope of fields and methods but with this interface the scope of the methods are now increased to public

Reply:
Good point.

I have removed those functions from the `IMapView` interface

```
onSelectionChanged(final ISelection selection);
void saveState();
```

But kept this one as it is used [here](https://github.com/mytourbook/mytourbook/blob/12a622074654e3f5799efb7135663d8ca80e528c/bundles/net.tourbook/src/net/tourbook/map25/Map25App.java#L805)

`void restoreState();`

Reply:
> But kept this one as it is used [here](https://github.com/mytourbook/mytourbook/blob/12a622074654e3f5799efb7135663d8ca80e528c/bundles/net.tourbook/src/net/tourbook/map25/Map25App.java#L805)
> 
> `void restoreState();`

but this will make the scope for this method in the other 2 classed also public

Reply:
Removed `void restoreState()`